### PR TITLE
Add maven wrapper for the library as well as for the maven example

### PR DESCRIPTION
**Pull request by dodgex**
see #1894

### CODE REVIEW

**Comment by dodgex** [comment]
I think we should not merge until we know what happened with #1905

**Comment by dodgex** [comment]
> [WARNING] Unknown file extension: /home/travis/build/androidannotations/androidannotations/AndroidAnnotations/mvnw
[WARNING] Missing header in: /home/travis/build/androidannotations/androidannotations/AndroidAnnotations/.mvn/wrapper/maven-wrapper.properties
[WARNING] Missing header in: /home/travis/build/androidannotations/androidannotations/AndroidAnnotations/mvnw.cmd

@WonderCsabo I think we should not add copyright headers here but instead see if we can tell the license check to ignore those files?

**Comment by WonderCsabo** [comment]
You are right. Please add these files to license check exclude list (it it easy).

Another thing: make sure the unix `mnw` files are committed to the repo with executable rights.

**Comment by dodgex** [comment]
You are right... `mvnw` shoud have `+x`. Any chance to do it from windows or do i need to checkout this branch on a linux system?

Will check this evening

**Comment by WonderCsabo** [comment]
I think you have to use a linux system for this.

**Comment by dodgex** [comment]
Looks like it is possible from windows too: http://blog.lesc.se/2011/11/how-to-change-file-premissions-in-git.html

**Comment by WonderCsabo** [comment]
Cool. Just make sure it is committed with the correct permissions.

File: .travis.yml [hunk 1]
```diff
@@ -26,12 +26,15 @@ before_script:
   - export TERM=dumb
 
 script:
-  - mvn -f AndroidAnnotations/pom.xml install -P travis
-  - mvn -f examples/maven/pom.xml install
+  - cd $TRAVIS_BUILD_DIR/AndroidAnnotations
+  - ./mvnw -f pom.xml install -P travis
```
Comment (WonderCsabo) [inline .travis.yml]: `-f pom.xml` is not needed, as we are in the correct dir.

File: .travis.yml [hunk 2]
```diff
@@ -26,12 +26,15 @@ before_script:
...
+  - cd $TRAVIS_BUILD_DIR/examples/maven
+  - ./mvnw -f pom.xml install
```
Comment (WonderCsabo) [inline .travis.yml]: `-f pom.xml` is not needed, as we are in the correct dir.

Comment (WonderCsabo) [inline .travis.yml]: `-f pom.xml` is not needed, as we are in the correct dir.

File: .travis.yml [hunk 3]
```diff
@@ -26,12 +26,15 @@ before_script:
...
+  - cd $TRAVIS_BUILD_DIR
   - ./examples/gradle/gradlew assembleDebug --build-file examples/gradle/build.gradle
 
 after_success:
-  - if [[ $TRAVIS_REPO_SLUG == 'androidannotations/androidannotations' && $TRAVIS_BRANCH == 'develop' && $TRAVIS_PULL_REQUEST == 'false' && $TRAVIS_JDK_VERSION == 'oraclejdk7' ]]; then mvn -f AndroidAnnotations/pom.xml -DskipTests -Dquiet -s settings.xml deploy ; fi
+  - if [[ $TRAVIS_REPO_SLUG == 'androidannotations/androidannotations' && $TRAVIS_BRANCH == 'develop' && $TRAVIS_PULL_REQUEST == 'false' && $TRAVIS_JDK_VERSION == 'oraclejdk7' ]]; then cd $TRAVIS_BUILD_DIR/AndroidAnnotations && ./mvnw -f pom.xml -DskipTests -Dquiet -s settings.xml deploy ; fi
```
Comment (WonderCsabo) [inline .travis.yml]: `-f pom.xml` is not needed, as we are in the correct dir.

**Comment by WonderCsabo** [comment]
@dodgex pls update the dev instructons on the wiki.

**Comment by dodgex** [comment]
Updated all usages of mvn with ./mvnw in wiki

**Pull request merged**